src/jmcomic/jm_downloader.py

```
@@ -269,6 +275,44 @@ def after_image(self, image: JmImageDetail, img_save_path):
+    def _invoke_features_for(self, when: str, **kwargs):
+        """
+        在指定钩子(when)中触发匹配的 Feature。
+
+        :param when: 当前钩子名，如 'after_album', 'after_photo'
+        :param kwargs: album, photo, downloader 等上下文
+        """
+        for feature, feature_from in self._feature_list:
+            if feature.should_invoke(feature_from, when):
+                try:
+                    feature.invoke(self.option, feature_from=feature_from, when=when, **kwargs)
+                except BaseException as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `BaseException`

(BLE001)